tests/encode-decode.test.ts

```diff
@@ -0,0 +1,43 @@
+import { describe, expect, it } from "@jest/globals";
+import { decodePermits, encodePermits, Permit, TokenType } from "../src";
+
+describe("Encoding / Decoding tests", () => {
+  it("Should properly encode and decode a list of permits", () => {
+    const permitErc20: Permit = {
+      beneficiary: "ubiquity",
+      deadline: "1",
+      networkId: 100,
+      nonce: "0x1",
+      owner: "0x1",
+      signature: "0x1",
+      tokenAddress: "0x1",
+      tokenType: TokenType.ERC20,
+      amount: 100,
+    };
+    const permitErc721: Permit = {
+      beneficiary: "ubiquity2",
+      deadline: "2",
+      networkId: 101,
+      nonce: "0x2",
+      owner: "0x2",
+      signature: "0x2",
+      tokenAddress: "0x2",
+      tokenType: TokenType.ERC721,
+      amount: 101.5,
```

Comment:
Just curious, isn't ERC721 (NFT reward) amount always supposed to equal 1?

Reply:
This one I do not have an answer for. There was no specific check that I have seen inside previous / other projects. If that's the case it would be worth adding checks around this. @0x4007 might know more

Reply:
@whilefoo @rndquu rfc

But yes intuitively I would say 1 is the correct answer.

Reply:
nice catch gitcoindev, yes non-fungible token, in pieces of 1

Reply:
I made the type enforcing "0" | "1" for ERC721 permits to avoid making this mistake again

Reply:
yes, it's always 1 (or 0)